packages/trpc/server/routers/viewer/eventTypes.tsx

```diff
@@ -434,15 +440,233 @@ export const eventTypesRouter = router({
       if (!user) {
         throw new TRPCError({ code: "INTERNAL_SERVER_ERROR" });
       }
-      return await ctx.prisma.eventType.findUnique({
+
+      const userSelect = Prisma.validator<Prisma.UserSelect>()({
```

Comment:
This logic is duplicate of getServerSideProps, let's reuse it. Maybe we can create a getEventType fn or something similar that is used by both getServerSideProps and tRPC.

The only difference I see is how 404 and other errors are handled.

Reply:
@Udit-takkar Can you take care of this as well.

Reply:
@hariombalhara done